cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -89,14 +89,34 @@ func RunPlan(flags *planFlags) error {
 		return err
 	}
 
-	// Define Local Etcd cluster to be able to retrieve information
-	etcdClient, err := etcdutil.NewStaticPodClient(
-		[]string{"localhost:2379"},
-		constants.GetStaticPodDirectory(),
-		upgradeVars.cfg.CertificatesDir,
-	)
-	if err != nil {
-		return err
+	var etcdClient etcdutil.Client
+
+	// Currently this is the only method we have for distinguishing
+	// external etcd vs static pod etcd
+	isExternalEtcd := len(upgradeVars.cfg.Etcd.Endpoints) > 0
+	if isExternalEtcd {
+		// Create external etcd client
+		client, err := etcdutil.NewClient(
```

Comment:
can't we do this for * in 1.11?

Reply:
Not quite, since we need to introspect the static pod manifest for the certs, which is what NewStaticPodClient does.